From 59b227cd5c0a30429318afc86e1dce0fec62639d Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Thu, 18 Oct 2018 20:12:38 +0300
Subject: [PATCH 122/741] staging: fsl-dpaa2/mac: read phy mode from device
 tree

commit 8c82cd63cb8b42545341bc838956481522038a81 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

If the a dpmac node defines its phy mode in the device tree using
the 'phy-mode' or the 'phy-connection-type' attributes this will take
precedence over the interface mode reported by the MC in the
dpmac attributes structure.

Signed-off-by: Ioana Ciornei <ioana.ciornei@nxp.com>
Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/staging/fsl-dpaa2/mac/mac.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/staging/fsl-dpaa2/mac/mac.c b/drivers/staging/fsl-dpaa2/mac/mac.c
index e913436bc1dc..a0fff3df4a25 100644
--- a/drivers/staging/fsl-dpaa2/mac/mac.c
+++ b/drivers/staging/fsl-dpaa2/mac/mac.c
@@ -466,7 +466,7 @@ static int dpaa2_mac_probe(struct fsl_mc_device *mc_dev)
 	struct dpaa2_mac_priv	*priv = NULL;
 	struct device_node	*phy_node, *dpmac_node;
 	struct net_device	*netdev;
-	phy_interface_t		if_mode;
+	int			if_mode;
 	int			err = 0;
 
 	dev = &mc_dev->dev;
@@ -555,6 +555,13 @@ static int dpaa2_mac_probe(struct fsl_mc_device *mc_dev)
 		goto probe_fixed_link;
 	}
 
+	if_mode = of_get_phy_mode(dpmac_node);
+	if (if_mode >= 0) {
+		dev_dbg(dev, "\tusing if mode %s for eth_if %d\n",
+			phy_modes(if_mode), priv->attr.eth_if);
+		goto phy_connect;
+	}
+
 	if (priv->attr.eth_if < ARRAY_SIZE(dpaa2_mac_iface_mode)) {
 		if_mode = dpaa2_mac_iface_mode[priv->attr.eth_if];
 		dev_dbg(dev, "\tusing if mode %s for eth_if %d\n",
@@ -565,6 +572,7 @@ static int dpaa2_mac_probe(struct fsl_mc_device *mc_dev)
 		goto probe_fixed_link;
 	}
 
+phy_connect:
 	/* try to connect to the PHY */
 	netdev->phydev = of_phy_connect(netdev, phy_node,
 					&dpaa2_mac_link_changed, 0, if_mode);
-- 
2.17.1

